Create README for RT-7.6: link bandwidth aggregation. (#3098)
openconfig/featureprofiles: feature/bgp/linkbandwidth_community/otg_tests/linkbandwidth_aggregation/README.md

# RT-7.6: BGP Link Bandwidth Community - Aggregation

## Summary

This test verifies Link-bandwidth (LBW) extended community aggregation feature by DUT.

## Testbed type

*  [`featureprofiles/topologies/atedut_2.testbed`](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_2.testbed)

## Procedure

### Test environment setup

    ```
                                          |         |
    [ 64x eBGP ] --- [ ATE Port 1 ] ----  |   DUT   | ---- [ ATE Port 2 ]
                                          |         |
    ```

### RT-7.6.1: Verify LBW aggregation to eBGP peer

* Configure DUT with routed ports on DUT.
* Configure 64x eBGP peers on ATE Port 1 interface in peering group UPSTREAM.
* Configure a single eBGP peer on ATE Port 2 interface in peering group DOWNSTREAM.
* Enable BGP LBW receive for peering group UPSTREAM.
* Enable BGP LBW send for peering group DOWNSTREAM. Enable Link Bandwidth aggregation feature.

> **TODO:** Link Bandwidth aggregation feature is not currently modeled in OC.

Reviewer: Is the intent to support [draft-ietf-bess-ebgp-dmz](https://datatracker.ietf.org/doc/draft-ietf-bess-ebgp-dmz/)?  Please add a reference link as it may help with clarity on the intent of the test.

I have opened a OpenConfig model update here for these paths.  Please reivew:
https://github.com/openconfig/public/pull/1131

> # RT-7.6: BGP Link Bandwidth Community - Aggregation

Reviewer: Suggest using same naming as ietf and OC:

```suggestion
# RT-7.6: BGP Link Bandwidth Community - Cumulative
```

## Summary

> This test verifies Link-bandwidth (LBW) extended community aggregation feature by DUT.

Reviewer: ```suggestion
This test verifies Link-bandwidth (LBW) extended community cumulative feature by DUT.
```

## Testbed type

*  [`featureprofiles/topologies/atedut_2.testbed`](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_2.testbed)

## Procedure

### Test environment setup

    ```
                                          |         |
    [ 64x eBGP ] --- [ ATE Port 1 ] ----  |   DUT   | ---- [ ATE Port 2 ]
                                          |         |
    ```

### RT-7.6.1: Verify LBW aggregation to eBGP peer

* Configure DUT with routed ports on DUT.
* Configure 64x eBGP peers on ATE Port 1 interface in peering group UPSTREAM.

> * Configure a single eBGP peer on ATE Port 2 interface in peering group DOWNSTREAM.

Reviewer: Style suggestion to place the pre-requisite configuration into a common section called Test Bed Setup Configuration (before the 7.6.1,2 subtests).   Pushing this config is not considered part of the 'test criteria', it's just a prerequisite.

Reviewer: ATE configuration should also be moved to the test bed setup configuration.

> * Enable BGP LBW receive for peering group UPSTREAM.
> * Enable BGP LBW send for peering group DOWNSTREAM. Enable Link Bandwidth aggregation feature.

Reviewer: This configuration is on the DUT, right?  

```suggestion
* Enable BGP LBW receive for peering group UPSTREAM on DUT.
* Enable BGP LBW send for peering group DOWNSTREAM on DUT.
* Enable Link Bandwidth aggregation feature on DUT.
```

> **TODO:** [Cumulative Link Bandwidth feature](https://datatracker.ietf.org/doc/draft-ietf-bess-ebgp-dmz/) is not currently modeled in OC. Cisco config example: `ebgp-send-extcommunity-dmz cumulative`

Reviewer: Add link to OC model pull request in progress: https://github.com/openconfig/public/pull/1131.

2. Advertise the same test prefix from ATE from all UPSTREAM peers with LBW community:
  * 32 peers - 10Mbps
  * 16 peers - 20Mbps
  * 8 peers - 40Mbps

>   * 8 peers - 80Mbps

Reviewer: Please move to testbed setup section.  This is useful because we aren't testing that the ATE can be configured.  So this fails, it's not a DUT test failure, it's a testbed infrastructure failure.

> 3. Verify that DUT advertises the test route to DOWNSTREAM eBGP peer with aggregated bandwidth community of 1600Mbps.

Reviewer: Minor tweak to be clear we want the test to verify by querying the ATE
```suggestion
3. Verify that ATE receives the test route to DOWNSTREAM eBGP peer with link bandwidth community set to 1600Mbps.
```

### RT-7.6.2: Verify LBW changes.
Using RT-7.6.1 set up conduct following changes:

1) Disable 32 peers advertising 10Mpbs bandwidth community.

> 2) Verify that DUT advertises the test route to Upstream peer with aggregated bandwidth community of 1280Mbps.

Reviewer: Here and throughout, rename "aggregated" to "cumulative"

3) Re-enable 32 peers advertising 10Mpbs bandwidth community.
4) Verify that DUT advertises the test route to Upstream peer with aggregated bandwidth community of 1600Mbps.

### RT-7.6.3: Verify LBW aggregation to iBGP peer

* Configure 64x eBGP peers on ATE Port 1 interface in peering group UPSTREAM.
* Configure a single iBGP peer on ATE Port 2 interface in peering group DOWNSTREAM.
* Enable BGP LBW receive for peering group UPSTREAM.
* Enable BGP LBW send for peering group DOWNSTREAM. Enable Link Bandwidth aggregation feature.

**TODO:** Link Bandwidth aggregation feature is not currently modeled in OC.

2. Advertise the same test prefix from ATE from all UPSTREAM peer with LBW community:
  * 32 peers - 10Mbps
  * 16 peers - 20Mbps
  * 8 peers - 40Mbps
  * 8 peers - 80Mbps

> 3. Verify that DUT advertises the test route to DOWNSTREAM iBGP peer with aggregated bandwidth community of 1600Mbps.

Reviewer: It's probably ok to specify changing the testbed configuration BGP peer types from eBGP to iBGP and repeat test 7.6.1.  (That's probably how someone would write the code)
